### PR TITLE
1256 Add support for more params in Inferers

### LINKED ISSUE #1256: Add *args **kwargs to inferers
**Is your feature request related to a problem? Please describe.**
Currently, our `inferer` methods only take 1 `inputs` as input data, some predictor may need additional parameters.
Will be useful to add `*args **kwargs` for inferers.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1256 .

### Description
This PR added support to pass more parameters to the `predictor` in `Inferers`.
And updated workflows based on ignite 0.4.2 new API(moved `to(device)` to `prepare_batch`):
https://github.com/pytorch/ignite/blob/v0.4.2/ignite/engine/__init__.py#L28
Already passed `workflow` and `workflow_gan` integration tests locally.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/engines/utils.py [hunk 1]
```diff
@@ -75,16 +75,29 @@ def get_devices_spec(devices: Optional[Sequence[torch.device]] = None) -> List[t
 
 
 def default_prepare_batch(
-    batchdata: Dict[str, torch.Tensor]
-) -> Union[Tuple[torch.Tensor, Optional[torch.Tensor]], torch.Tensor]:
+    batchdata: Dict[str, torch.Tensor],
+    device: Optional[Union[str, torch.device]] = None,
```
Comment (wyli) [inline monai/engines/utils.py:79]: there's a `non_blocking: bool = False` in ignite, why not add the option here as well?

File: monai/inferers/utils.py [hunk 2]
```diff
@@ -70,6 +72,8 @@ def sliding_window_inference(
             By default the device (and accordingly the memory) of the `inputs` is used. If for example
             set to device=torch.device('cpu') the gpu memory consumption is less and independent of the
             `inputs` and `roi_size`. Output is on the `device`.
+        args: optional args to be passed to ``network``.
```
Comment (wyli) [inline monai/inferers/utils.py:75]:
```suggestion
        args: optional args to be passed to ``predictor``.
```

File: monai/inferers/utils.py [hunk 3]
```diff
@@ -70,6 +72,8 @@ def sliding_window_inference(
...
+        kwargs: optional keyword args to be passed to ``network``.
```
Comment (wyli) [inline monai/inferers/utils.py:76]:
```suggestion
        kwargs: optional keyword args to be passed to ``predictor``.
```

File: monai/engines/evaluator.py [hunk 4]
```diff
@@ -171,19 +171,16 @@ def _iteration(self, engine: Engine, batchdata: Dict[str, torch.Tensor]) -> Dict
         """
         if batchdata is None:
             raise ValueError("Must provide batch data for current iteration.")
-        inputs, targets = self.prepare_batch(batchdata)
-        inputs = inputs.to(engine.state.device)
-        if targets is not None:
-            targets = targets.to(engine.state.device)
+        inputs, targets, args, kwargs = self.prepare_batch(batchdata, engine.state.device)
```
Comment (wyli) [inline monai/engines/evaluator.py:174]: I feel this becomes confusing if the user is already familiar with ignite's prepare batch (returns two items).
could we make it flexible with something like:
```suggestion
        batch = self.prepare_batch(batchdata, engine.state.device)
        if len(batch) == 2:
             inputs, targets = batch
        else:
             inputs, targets, args, kwargs = batch
```
or
```suggestion
        inputs, targets, *args = self.prepare_batch(batchdata, engine.state.device)
```

File: monai/engines/utils.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/engines/utils.py:79]: Sure, added `non_blocking` support to all the engines, to be consistent with the latest ignite.
Thanks.

File: monai/inferers/utils.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/inferers/utils.py:75]: updated in the latest commit.
Thanks.

File: monai/inferers/utils.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/inferers/utils.py:76]: updated in the latest commit.
Thanks.

File: monai/engines/evaluator.py (hunk 4 above)
Comment (Nic-Ma) [reply monai/engines/evaluator.py:174]: Sure, updated to suggestion 1 in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I updated the PR according to your comments.
Could you please help review it again?

Thanks.

**Review by wyli — APPROVED** [review]
thanks

**Pull request merged**